lib/block-supports/duotone.php

```diff
@@ -341,25 +301,74 @@ function ( $selector ) use ( $duotone_id ) {
 
 	<?php
 
-	$duotone = ob_get_clean();
+	$svg = ob_get_clean();
+
+	// Clean up the whitespace so it can be used in a data uri.
+	$svg = preg_replace( "/(\r|\n|\t)+/", ' ', $svg );
+	$svg = preg_replace( '/> </', '><', $svg );
+	$svg = trim( $svg );
+
+	$data_uri = 'data:image/svg+xml,' . $svg . '#' . $filter_id;
+
+	// All the variables are already escaped above, so we're not calling esc_url() here.
+	return "url('" . $data_uri . "')";
```

Comment:
This function may benefit from unit tests to make sure we're encoding the uri string properly and that we're not exposing potential xss issues when given some malformed data.

🤔 did we really not need to encode any characters here? In the data URI spec, I'm pretty sure that there are some reserved characters here.

<img width="805" alt="CleanShot 2021-09-09 at 10 47 27@2x" src="https://user-images.githubusercontent.com/1270189/132736685-56466ca0-985c-44cc-b9b9-1f18c1cb0398.png">

I put up a quick code pen and see that this is working for Chrome and FF but not Safari:

https://codepen.io/gwwar/pen/mdwWXwM

<img width="1646" alt="CleanShot 2021-09-09 at 11 23 45@2x" src="https://user-images.githubusercontent.com/1270189/132741796-11945a84-f867-45aa-a6f6-8b2a1f080d3a.png">

Reply:
> :thinking: did we really not need to encode any characters here? In the data URI spec, I'm pretty sure that there are some reserved characters here.

Huh, none of the examples I saw ever did any escaping, but it looks like you're right about the spec. I guess this is just one of those things that browsers automatically try to fix.

`esc_url()` seems to strip out the data URI entirely. Is `urlencode()` enough here? `esc_attr()` above should already cover the XSS vectors, right?

> I put up a quick code pen and see that this is working for Chrome and FF but not Safari

:slightly_frowning_face: safari. 

Thanks for testing Safari on everything! I tend to forget about it when I do all my development on Linux. Looks like I'll have to break out the MacBook Pro for this one. :slightly_smiling_face:

Reply:
Looks like it doesn't matter how we encode the SVG in Safari. It just doesn't work in the CSS.

<img width="1552" alt="image" src="https://user-images.githubusercontent.com/5129775/132757453-1282bccd-1d33-4c4c-af76-c5e261d5bd4e.png">

https://codepen.io/ajlende/pen/xxrqjRy?editors=1100

Reply:
Looks like it must be a browser bug https://bugs.webkit.org/show_bug.cgi?id=211617 I also see the same issue with the Safari Technology Preview. We can maybe add a comment noting our use case on the issue.

Note that even without the inline SVG issue, it looks like Safari has issues with applying the filter appropriately via CSS https://bugs.webkit.org/show_bug.cgi?id=114303 and on videos https://bugs.webkit.org/show_bug.cgi?id=184601

<img width="1230" alt="CleanShot 2021-09-09 at 13 55 35@2x" src="https://user-images.githubusercontent.com/1270189/132761355-56572d0c-4689-411d-8f5c-daa47ebd6d3f.png">